### ISSUE TITLE
Releases outside of App Store?

### ISSUE THREAD

**Issue by Miaourt** [passepartoutvpn/passepartout-apple#231]
Hi ! I don't have AppleID and so really can't use App Store but would love to use passpartout too, would it be possible to have a version outside of App Store ?

**Comment by keeshux** [comment]
The app relies on a VPN framework that enforces App Store distribution. So, unfortunately, this is not possible.

**Comment by Miaourt** [comment]
Is this "VPN framework" this one https://github.com/passepartoutvpn/tunnelkit ?
Shall I raise an issue down there to investigate this, or does it simply mean you're not interested into making something that works outside the App Store ?

**Comment by keeshux** [comment]
Nope, framework is [NetworkExtension](https://developer.apple.com/documentation/networkextension).

**Comment by Miaourt** [comment]
Hello, sorry to resurrect the issue!
Took a look at Apple's docs on NetworkExtension, and now they do [provide a couple of step](https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_networking_networkextension) for app released outside the app store... so NetworkExtension is now usable without being distributed thru the app store!
Could releasing it on Github releases be considered again? 👀

**Comment by keeshux** [comment]
@Miaourt thanks for bringing this up, this is awesome! Another "meta" issue with such a release, however, is in-app purchases.

**Comment by Miaourt** [comment]
Awesome that you're considering it again! 

Yeah it might be an issue and maintaining a "license" server somewhere (to do like Viscosity for example) where we additional features is, while not that difficult, a non-zero additional work I guess.

If you'd consider making a licensing server open source I'll be glad to help on this point.

**Comment by Miaourt** [comment]
To be more clear, this way of activating additional features would be lock to the version outside of App Store, to not conflict with Apple's policies.

**Comment by Miaourt** [comment]
Hello @keeshux! Happy new year.
Did you had some time to think about it?

**Comment by keeshux** [comment]
Part of the 2024 roadmap (still in the works).

**Comment by keeshux** [comment]
@Miaourt FYI, I started working on this. Let's see how it turns out.

**Comment by Miaourt** [comment]
@keeshux awesome! If you need help with anything, be it just testing it, I have some free time.
Just for curiosity, how do you plan to handle the in-app purchases that previously bothered you?

**Comment by keeshux** [comment]
I'll start by omitting the paid features and will see where the thing goes later.

**Comment by keeshux** [comment]
@Miaourt check out the artifacts at the bottom of this workflow:

https://github.com/passepartoutvpn/passepartout/actions/runs/15281819309

**Comment by Miaourt** [comment]
Did some test with a mullvad VPN, seems to work 🎉

Some note I encountered:
- When installing, some weirdo like me install apps in `~/Applications` (so they are less privilegied by default), other VPN clients warn you that it can't work in this location and that you need to install it int `/Applications` for the network extension to work
- tried to load a mullvad's IPv6 config, got greeted by a `Module WireGuard is malformed. Unable to parse file. macAlertEndpointInvalid Peer’s endpoint must be of the form ‘host:port’ or ‘[host]:port’.`

Otherwise, nice work ! I'm sadly not using much VPN on my macbook since 2023, so I can't really dailydrive it to reports issues in the following days, but feel free to poke me if needed!
